lib/main.dart

```diff
@@ -10,17 +12,30 @@ void main() {
   runApp(SafeArea(child: ProviderScope(child: MyApp())));
 }
 
-class MyApp extends StatelessWidget {
+class MyApp extends StatefulWidget {
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+
+class _MyAppState extends State<MyApp> {
+  bool _isDark = false;
+
+  void _changeTheme() {
+    setState(() {
```

Comment:
Nice work. But this project uses providers. Please shift `setState` to provider.